ca_driver/src/create_driver.cpp

```diff
@@ -222,6 +224,16 @@ void CreateDriver::undockCallback(const std_msgs::EmptyConstPtr& msg)
   robot_->setMode(create::MODE_FULL);
 }
 
+void CreateDriver::defineSongCallback(const ca_msgs::DefineSongConstPtr& msg)
+{
+  robot_->defineSong(msg->song, msg->length, &(msg->notes.front()), &(msg->durations.front()));
```

Comment:
Done. It would be nice to add more error checking to libcreate create::Create::defineSong. Perhaps that is a 2nd PR.